app/models/content_item.rb

```diff
@@ -171,12 +171,20 @@ def router_rendering_app
     rendering_app || "government-frontend"
   end
 
-  def viewable_by?(user_uid)
-    authorised_user_uids.empty? || authorised_user_uids.include?(user_uid)
+  def viewable_by_user_id?(user_id)
+    auth_user_ids.include?(user_id)
   end
```

Comment:
Thinking out load a bit but I wonder if we could encapsulate more of the viewing logic in the model rather than in the controller. Though that might be a bad idea if we have to return different responses based on different scenarios.

We could have a method:

```
def viewable?(auth_bypass_id, user_id, organisation_id)
  return auth_bypass_ids.include?(auth_bypass_id) if auth_bypass_id
  return true if auth_user_ids.empty? && auth_org_ids.empty?

  auth_user_ids.include?(user_id) || auth_org_ids.include?(organisation_id)
end
```

which could reduce the number of methods and logic scattered around, and provide a more concrete thing to test to make logic expectations clear.

Reply:
Again yup - we've had a go at writing the viewable? method in the content item model.